v11.7 ESR Changelog (#8903)
mattermost/docs: source/product-overview/mattermost-v11-changelog.md

```{include} common-esr-support-upgrade.md
```

(release-v11.7-extended-support-release)=
## Release v11.7 - [Extended Support Release](https://docs.mattermost.com/product-overview/release-policy.html#release-types)

**Release day 2026-05-15**

```{Attention}
**Breaking Changes**
FIPS builds require a minimum of 14 characters for passwords, atmos/camo proxy configuration, and shared channel secrets. Shorter passwords for existing users will no longer be valid and require a password reset. Non-FIPS builds are unaffected.
```

#### Database Schema Changes
 - TBD

#### config.json
New setting options were added to ``config.json``. Below is a list of the additions and their default values on install. The settings can be modified in ``config.json``, or the System Console when available.
 - **Changes to Enterprise Advanced plan:**
   - Under ``ExperimentalSetting`` in ``config.json``, added ``EnableWatermark`` configuration setting to add watermarking toggle in the server.
 - **Changes to Enterprise plans:**

>    - Under ``PasswordSettings`` in ``config.json``, added ``​​PasswordFIPSMinimumLength`` configuration setting to require FIPS builds a minimum of 14 characters for passwords, atmos/camo proxy configuration, and shared channel secrets.

Reviewer: ```suggestion
   - Under ``PasswordSettings`` in ``config.json``, added ``​​PasswordFIPSMinimumLength`` configuration setting to require a minimum of 14 characters for passwords and for atmos/camo proxy and shared channel secrets.
```

Author: Thanks!

 - Channel membership changes are now reliably synchronized between connected workspaces, even when a remote server is temporarily offline. Previously, membership removals could be missed during outages.
 - Added an "Open in new tab" button to the **Product Switcher** menu.
 - Substring matching is now allowed when searching channel members in the member sub-panel in a channel.
 - Improved accessibility of thread list menus.
 - Channel banners are now shown in thread views.
 - Improved autocomplete while typing in Korean and using Firefox.
 - Updated license renewal and expiry notification emails with refreshed branding, copy, and layout.
 - Added the ability to open channels in a separate popout window, with full channel and right-hand side functionality.
 - Dropped support for JS features required by browsers over three years old.
 - Renamed user-visible references from "Custom Profile Attributes" to "User Attributes" across the admin console, error messages, and server translations.
 - Added the ability to handle from which remotes a channel is shared from the channel settings user interface.

#### Administration

>  - Removed direct dependency on ``blang/semver/v4``.

Reviewer 2: Not worth to mention, I think

Author: Thanks!

 - Added new API endpoints ``PUT /api/v4/system/e2e/ai_bridge``, ``GET /api/v4/system/e2e/ai_bridge``, and ``DELETE /api/v4/system/e2e/ai_bridge`` for E2E testing of AI features. These endpoints are only accessible when ``EnableTesting`` is true. Refactored internal AI-related logic to use a new ``AgentsBridge`` interface for improved testability. Added new DTOs in ``server/public/model`` for AI bridge information and test helpers.
 - Added a new API endpoint ``PUT /api/v4/channels/{channel_id}/members`` that sets the complete membership of a channel in a single call. The endpoint accepts a JSON object with ``members`` (desired user IDs) and an optional ``channel_admins`` (user IDs to designate as channel admins). The server computes the diff against current membership, adds or removes users as needed, and reconciles admin roles. When ``channel_admins`` is omitted, existing admin roles are preserved. Results are streamed back as NDJSON for progress tracking. Requires system admin permissions.

>  - Added three new plugin APIs for shared channel sync: ``SendSharedChannelSyncMsg``, ``SendSharedChannelAttachmentSyncMsg``, and ``SendSharedChannelProfileImageSyncMsg``. These allow plugins acting as shared channel remotes to sync posts, reactions, users, file attachments, and profile images into Mattermost, complementing the existing outbound ``OnSharedChannels`` hooks.

Reviewer: ```suggestion
 - Added three new plugin APIs for shared channel sync: ``ReceiveSharedChannelSyncMsg``, ``ReceiveSharedChannelAttachmentSyncMsg``, and ``ReceiveSharedChannelProfileImageSyncMsg``. These allow plugins acting as shared channel remotes to sync posts, reactions, users, file attachments, and profile images into Mattermost, complementing the existing outbound ``OnSharedChannels`` hooks.
```

 - Improved autocomplete while typing in Korean and using Firefox.
 - Updated license renewal and expiry notification emails with refreshed branding, copy, and layout.
 - Added the ability to open channels in a separate popout window, with full channel and right-hand side functionality.
 - Dropped support for JS features required by browsers over three years old.
 - Renamed user-visible references from "Custom Profile Attributes" to "User Attributes" across the admin console, error messages, and server translations.
 - Added the ability to handle from which remotes a channel is shared from the channel settings user interface.

#### Administration
 - Updated dependency versions for server and public services.
 - Added managed channel categories for Channel Admins to enforce sidebar organization across teams.
 - Migrated access control policies from v0.2 to v0.3, replacing the wildcard action with explicit action types to support multi-action policies.
 - Added team-level ABAC membership policies, allowing Team Admins to create and manage attribute-based access policies for private channels within their team directly from a new Membership Policies tab in the Team Settings Modal.

>  - Added permission policies.

Reviewer 3: ```suggestion
 - Added permission policies to manage file attachment download/upload actions.
 - Policy editor is improved for "Multi Select" attribute types to use more intuitive operators.
```
